src/context/WalletProvider/NativeWallet/components/NativeCreate.tsx

```diff
@@ -59,14 +62,14 @@ export const NativeCreate = ({ history, location }: NativeSetupProps) => {
   useEffect(() => {
     ;(async () => {
       try {
-        const vault = await getVault()
+        const vault = await getVault(location.state?.mnemonic)
```

Comment:
We don't want to pass the mnemonic around nor set it in state, we'll want to use the mnemonic to create a wallet vault and pass the vault around.

I'm not sure the wallet vault is serializable, so I'm not sure if it'll work setting it into state.